### PR TITLE
Add how to create custom domains with kn domain

### LINKED ISSUE #3496: document Client CLI kn domain
The new version of the `kn` CLI container a new command `domain`

The info is included in the release notes but it should properly be documented in the website

It was suggested by @abrennan89 to open an issue in the PR for the release notes blod [here](https://github.com/knative/docs/pull/3486#discussion_r617561269)
>@csantanapr can you please open a docs issue for kn domain to get it properly documented in the site so it's not only explained here?

/assign @rhuss

### PR DESCRIPTION
**Pull request by rhuss**
## Proposed Changes 

- Add a description of how to leverage `kn domain` for managing custom domains
- Extended YAML description for `CustomDomain` include the `tls:` section for enabling HTTPS

Note, that I'm not sure about the formatting as I add the `kn` usage in the middle of a numbered list and am not sure how it will be rendered. Any help for fixing the formatting is appreciated. 

Fixes #3496

### CODE REVIEW

**Comment by snneji** [comment]
Hi @rhuss, some of the formatting fixes will need to span lines that you have not edited in the PR so it will be difficult for me to use the suggestion feature to suggest the correct formatting. 

Would it be OK to merge the PR and then open a cleanup issue for us to fix the formatting and style for you? Or would you prefer me to leave comments on the PR for you to fix them yourself?

**Comment by rhuss** [comment]
@snneji happy to merge it now and hand it over to you to fix the formatting. So maybe give it a quick look about whether the content is fine so far ?

File: docs/developer/serving/services/custom-domains.md [hunk 1]
```diff
@@ -86,5 +91,19 @@ DomainMappings in that namespace to use the domain name.
...
 
+=== "kn CLI"
+
+    ```
+    kn domain create <domain-name> --ref <target> --tls <tls-secret> --namespace <namespace>
+    ```
+
+    Where:
+    - `<domain-name>` is the domain name that you want to map a Service or Route to.
+    - `<target>` is the name of the Service or Route that is mapped to the domain. You can use a prefix `ksvc:` or `kroute:` to specify whether to map the domain to a Knative Service or Route. If no prefix is given, `ksvc:` is assumed. Additional, you can a `:namespace` suffix to point to a Service or Route in a different namespace than this Domain mapping. E.g. `mysvc` maps to a Service `mysvc` in the same namespace as this mapping, while `kroute:myroute:othernamespace` maps to a Route `myroute` in namespace `othernamespace`.
```
Comment (snneji) [inline docs/developer/serving/services/custom-domains.md:102]:
```suggestion
    - `<target>` is the name of the Service or Route that is mapped to the domain. You can use a prefix `ksvc:` or `kroute:` to specify whether to map the domain to a Knative Service or Route. If no prefix is given, `ksvc:` is assumed. Additionally, you can use a `:namespace` suffix to point to a Service or Route in a different namespace than this Domain mapping. E.g. `mysvc` maps to a Service `mysvc` in the same namespace as this mapping, while `kroute:myroute:othernamespace` maps to a Route `myroute` in namespace `othernamespace`.
```

**Review by snneji — CHANGES_REQUESTED** [review]
Looks OK to merge to me. Just a small fix here and the rest of the fixes can be dealt with in the cleanup issue. 

One other thing - does this PR need to be cherry picked? It seems to be based on an issue that was first created for v0.22.

**Comment by rhuss** [comment]
`kn` supports the `domain` command since 0.22, not sure about the `tls:` option though. @julz do you know when `tls:` was introduced to DomainMapping ? Was it there from the beginning ?

**Comment by julz** [comment]
wasn't there in the beginning, looks like it [landed in 0.24](https://github.com/knative/serving/releases/tag/v0.24.0) (same time DM went beta)

**Review by snneji — APPROVED** [review]

**Comment by snneji** [comment]
Issue to fix formatting: https://github.com/knative/docs/issues/4225

**Pull request merged**